packages/ember-glimmer/lib/component.js

```diff
@@ -104,9 +200,282 @@ const Component = CoreView.extend(
+        Hi!
+      {{/foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{#if hasBlock}}
+        This will be printed because a block was provided
+        {{yield}}
+      {{/if}}
+      ```
+
+      This helper accepts an argument with the name of the block we want to check the presence of.
+      This is useful for checking for the presence of the optional inverse block in components.
+
+      ```hbs
+      {{! templates/application.hbs }}
+      {{#foo-bar}}
+        Hi!
+      {{else}}
+        What's up?
+      {{/foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{yield}}
+      {{#if (hasBlock "inverse")}}
+        {{yield to="inverse"}}
+      {{else}}
+        How are you?
+      {{/if}}
+
+      ```
+      @public
+      @property hasBlock
+      @param {String} [blockName="default"] The name of the block to check presence of.
+      @returns Boolean
+      @since 1.13.0
+     */
+
+    /**
+      Returns true when the component was invoked with a block parameter
+      supplied.
+
+      Example (`hasBlockParams` will be `false`):
+      ```hbs
+      {{! templates/application.hbs }}
+      {{#foo-bar}}
+        No block parameter.
+      {{/foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{#if hasBlockParams}}
+        This will not be printed, because no block was provided
+        {{yield this}}
+      {{/if}}
+      ```
+
+      Example (`hasBlockParams` will be `true`):
+
+      ```hbs
+      {{! templates/application.hbs }}
+      {{#foo-bar as |foo|}}
+        Hi!
+      {{/foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{#if hasBlockParams}}
+        This will be printed because a block was provided
+        {{yield this}}
+      {{/if}}
+      ```
+
+      @public
+      @property hasBlockParams
+      @returns Boolean
+      @since 1.13.0
+   */
+   /**
+     Enables components to take a list of parameters as arguments.
+     For example, a component that takes two parameters with the names
+    `name` and `age`:
+
+    ```javascript
+    let MyComponent = Ember.Component.extend;
+    MyComponent.reopenClass({
+      positionalParams: ['name', 'age']
+    });
+    ```
+
+    It can then be invoked like this:
+
+    ```hbs
+    {{my-component "John" 38}}
+    ```
+
+    The parameters can be referred to just like named parameters:
+
+    ```hbs
+    Name: {{attrs.name}}, Age: {{attrs.age}}.
+    ```
+
+    Using a string instead of an array allows for an arbitrary number of
+    parameters:
+
+    ```javascript
+    let MyComponent = Ember.Component.extend;
+    MyComponent.reopenClass({
+      positionalParams: 'names'
+    });
+    ```
+
+    It can then be invoked like this:
+
+    ```hbs
+    {{my-component "John" "Michael" "Scott"}}
+    ```
+
+    The parameters can then be referred to by enumerating over the list:
+
+    ```hbs
+    {{#each attrs.names as |name|}}{{name}}{{/each}}
+    ```
+
+    @static
+    @public
+    @property positionalParams
```

Comment:
Should we remove `positionalParams` for the same reason as hasBlock/hasBlockParams?